File: app/src/main/java/com/rsupport/mobile1/test/application/BaseApplication.java
package com.rsupport.mobile1.test.application;

import android.app.Application;
import android.os.Build;

import com.rsupport.mobile1.test.utils.NetworkConnectionCheck;

public class BaseApplication extends Application {
    public static NetworkConnectionCheck networkConnectionCheck;
    private String TAG = BaseApplication.class.getSimpleName();

    public BaseApplication() {
    }

    @Override
    public void onCreate() {
        super.onCreate();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            if (networkConnectionCheck == null) {
                networkConnectionCheck = new NetworkConnectionCheck(getApplicationContext());
                networkConnectionCheck.register();
            }
        }
    }

}

File: app/src/main/java/com/rsupport/mobile1/test/application/BaseApplication.kt
package com.rsupport.mobile1.test.application


import android.app.Application
import android.os.Build
import com.rsupport.mobile1.test.utils.NetworkConnectionCheck

class BaseApplication : Application() {
    private val TAG = BaseApplication::class.java.simpleName
    override fun onCreate() {
        super.onCreate()
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            if (networkConnectionCheck == null) {
                networkConnectionCheck = NetworkConnectionCheck(
                    applicationContext
                )
                networkConnectionCheck!!.register()
            }
        }
    }

    companion object {
        var networkConnectionCheck: NetworkConnectionCheck? = null
    }
}